components/automate-cli/pkg/verifyserver/services/startmockserverservice/startmockserverservice_test.go

```diff
@@ -0,0 +1,270 @@
+)
+
+func TestStartMockServer(t *testing.T) {
+	servers := startmockserverservice.StartMockServerService{}
+
+	t.Run("Start TCP server", func(t *testing.T) {
+		cfg := models.StartMockServerRequestBody{
+			Protocol: "tcp",
+			Port:     8000,
+		}
+
+		server, err := servers.StartMockServer(cfg)
+		require.NoError(t, err)
+		require.NotNil(t, server)
+
+		// Check that the server is listening on the correct port
+		require.Equal(t, cfg.Protocol, server.ListenerTCP.Addr().Network())
+		require.Equal(t, cfg.Port, server.ListenerTCP.Addr().(*net.TCPAddr).Port)
+
+		server2, err := servers.StartMockServer(cfg)
+		require.Error(t, err)
+		require.Nil(t, server2)
+		// fmt.Print()
+		// Stop the server and check that it was closed correctly
+		err = server.ListenerTCP.Close()
+		require.NoError(t, err)
+		if server2 != nil {
+			err = server2.ListenerTCP.Close()
+			require.Error(t, err)
+		}
+	})
+
+	t.Run("Start UDP server", func(t *testing.T) {
+		cfg := models.StartMockServerRequestBody{
+			Protocol: "udp",
+			Port:     8001,
+		}
+		server, err := servers.StartMockServer(cfg)
+		require.NoError(t, err)
+		require.NotNil(t, server)
+
+		// Check that the server is listening on the correct port
+		require.Equal(t, cfg.Protocol, server.ListenerUDP.LocalAddr().Network())
+		require.Equal(t, cfg.Port, server.ListenerUDP.LocalAddr().(*net.UDPAddr).Port)
+
+		fmt.Println(server.ListenerUDP.LocalAddr())
+		// Stop the server and check that it was closed correctly
+		err = server.ListenerUDP.Close()
+		require.NoError(t, err)
+	})
+
+	t.Run("Start HTTPS server", func(t *testing.T) {
+
+		cfg := models.StartMockServerRequestBody{
+			Protocol: "https",
+			Port:     8002,
+			Cert:     SERVER_CERT,
+			Key:      SERVER_KEY,
+		}
+		server, err := servers.StartMockServer(cfg)
+		require.NoError(t, err)
+		require.NotNil(t, server)
+
+		// Check that the server is listening on the correct port
+		require.Equal(t, fmt.Sprintf(":%d", cfg.Port), server.ListenerHTTP.Addr)
+		// require.Equal(t, cfg.Protocol, server.ListenerHTTP.Addr)
+
+		// Stop the server and check that it was closed correctly
+		err = server.ListenerHTTP.Close()
+		require.NoError(t, err)
+	})
+
+	t.Run("Unsupported protocol", func(t *testing.T) {
+		_, err := servers.StartMockServer(models.StartMockServerRequestBody{
+			Port:     8003,
+			Protocol: "http",
+			Cert:     "",
+			Key:      "",
+		})
+
+		require.Error(t, err)
+		require.Equal(t, "unsupported protocol", err.Error())
+	})
+}
+
+func TestHandleTCPRequest(t *testing.T) {
+	// create a new StartMockServerService
+	service := &startmockserverservice.StartMockServerService{}
+
+	t.Run("For healthy connetion", func(t *testing.T) {
```

Comment:
use your code to create server instead of creating here

Reply:
Done